core/src/main/scala/stryker4s/run/EtaCalculator.scala

```diff
@@ -0,0 +1,54 @@
+package stryker4s.run
+
+import grizzled.slf4j.Logging
+
+import scala.concurrent.duration._
+import language.postfixOps
+import scala.collection.mutable
+
+class EtaCalculator(amountOfMutants: Int) extends Logging {
```

Comment:
In general, I get the feeling that this class is doing a lot of stuff it's not responsible for.

1. prettyPrintDuration could be an extension method on Duration so I expected an implicit class for the duration in that case.
2. No way the ETACalculator is responsible for invoking the "fun" because that seems to be the "runMutant" function
3. keeping track of the duration a mutation run took is no responsibility for this class. the runMutant function could result in mutationResult which includes a duration.

When considering option 3 this class could then simply queue the last 10 and calculate an avg based on the runtime. This avg is then reported as a duration, which can then be used in the mutantrunner to be prettyPrinted using the extentionMethod.

Reply:
To be fair, I had a feeling I wasn't thinking according to the correct paradigm here either...

So please help me understand a few things. :')

1. I can get behind this one... I suppose conversion/parsing logic like this is cleaner in a seperate class. The reason I didn't do it initially was because: "it's only used here, just make it a private function".
1. It is basically calling the `runMutant` function, true. But I felt like the alternative is doing something awful like:
```
EtaCalculator.start()
runMutant(...)
EtaCalculator.stop()
```
How do I implement this neatly then? :/ 

3.  Giving `runMutant` responsibility to time itself means we need to implement the same timing logic in every implementation of `MutantRunner` which would result in duplicate code. (Or we would have to introduce another function that calls `runMutant` and wraps it inside a `MutantRunResult` with the duration?)